### DISCUSSION TITLE
Does anyone have a 128x64 I can see?

### DISCUSSION THREAD

**Discussion by BillLehecka** [MLB-LED-Scoreboard/mlb-led-scoreboard#495] [Show and tell]
I just invested in a 128x64 to upgrade my current 64x32 setup. I mainly did this so I can show more information like pitch count, pitch type, etc. It'll be coming next week. In the meantime, I've been looking far and wide for an example of a 128x64 setup, but I haven't seen any in the wild. Can someone post what they did?

**Comment by twfarley** [comment]
I’ll post mine this evening. This would be a great discussion for Discord. On Jul 18, 2023, at 8:33 AM, Bill Lehecka ***@***.***> wrote:﻿
I just invested in a 128x64 to upgrade my current 64x32 setup. I mainly did this so I can show more information like pitch count, pitch type, etc. I'll be coming next week. In the meantime, I've been looking far and wide for an example of a 128x64 setup, but I haven't seen any in the wild. Can someone post what they did?

**Comment by ty-porter** [comment]
You can always emulate! The beauty of doing that is once you get your new board, it should be plug-n-play with the custom configuration you've come up with.

**Comment by BillLehecka** [comment]
Ah yes, emulation. I can do that directly on my Pi right?

**Reply by BillLehecka** [reply]
So I got my 128x64 board today, I went in and changed my service code to reflect the cols of 128 and the rows of 64... and this happens:
![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/3dc28df8-5edb-4206-b861-64a8173793c5)

Any clue as to why this would be happening. Here are the arguments I'm passing to start it:

Environment="SCOREBOARD_ARGS=--led-gpio-mapping=adafruit-hat-pwm --led-brightness=40 --led-slowdown-gpio=2 --led-cols=128 --led-rows=64"

Or is this an issue with my coordinates file?

**Reply by BillLehecka** [reply]
Additional note: I checked it in emulation and didn't seem to have any issues.

**Reply by WardBrian** [reply]
Have you played around any with the `--led-multiplexing` flag?

**Reply by BillLehecka** [reply]
OK. Read up on the rgb-led-matrix code base. And fiddled around with led-multiplexing. Tried led-multiplexing=1 and now it's just two chunks of lines not showing. 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/6fbb4363-1491-4877-9705-d0d0e39d1d6f)

The more I research this, the more I feel like I may need to get the ol' soldering gun out. Unless there's something I'm missing in the arguments I should be passing. Perhaps a different multiplexer?

Or perhaps just try all 17 to see what sticks?

**Comment by twfarley** [comment]
Did you solder the E pin on the HAT?On Aug 3, 2023, at 1:00 PM, Bill Lehecka ***@***.***> wrote:﻿
OK. Read up on the rub-led-matrix code base. And fiddled around with led-multiplexer. Tried led-multiplexer=1 and now it's just two chunks of lines not showing.

The more I research this, the more I feel like I may need to get the ol' soldering gun out. Unless there's something I'm missing in the arguments I should be passing.

**Reply by BillLehecka** [reply]
This one? No. It is the Adafruit Bonnet. Where should be be soldered to?

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/9b148900-8528-4ca6-9ad8-2848f0cf9386)

I did the old 4 to 18 solder job when I first built this in 2020 with the 64x32 board.

**Comment by twfarley** [comment]
Check out https://learn.adafruit.com/adafruit-rgb-matrix-bonnet-for-raspberry-pi/driving-matrices

If you're using a 64x64 RGB matrix and either a Bonnet or a Rev C HAT, use your soldering iron to melt a blob of solder on the bottom solder jumper so the middle pad is 'shorted' to 8. (This is compatible with 64x64 matrices in the Adafruit store. For 64x64 matrices from other sources, you might need to use 16 instead, check the datasheet.)

**Reply by BillLehecka** [reply]
So this should work with a 128x64 board then, since it seems my board can handle 128 columns just fine, but is struggling with the 64 rows. 

I'll take a look at it when I get back from my weekend trip. Thanks for the feedback.

**Reply by pjockey** [reply]
I'm using 128X64 with both this, mlb, and the nhl. Both work just find. My board is never turned off. 24/7/365!!!

**Reply by pjockey** [reply]
I purchased a couple of 128x64 from AliExpress and one works and one is still a work in progress. The attached image is the one that worked, but it is no longer availbel, go figure.
<img width="935" alt="Screen Shot 2023-08-15 at 18 44 34" src="https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/1387701/0261e0fc-e0e5-4267-9fb4-60dc7844f6c6">

**Reply by BillLehecka** [reply]
I had an ordered cancelled from AliExpress when I tried for a board before. That being said, I found one that is HUB75 (not HUB75E) so I'm going to try this. It said it was shipped this morning, so crossing fingers. 

https://m.aliexpress.us/item/3256804188393351.html?spm=a2g0n.order_detail.order_detail_item.4.ffccf19cYAugOX&gatewayAdapt=glo2usa

Question: will having E and 8 soldered together now be a problem if I switch back to a HUB75 board?

**Comment by c2mfj** [comment]
you need this solder to go higher than 32 rows.

**Comment by BillLehecka** [comment]
OK. Soldered E to 8. Still alternates 16 on, 16 off. ![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/11736b35-7e2a-480b-88e1-d1ef0b6c5f54)

The one thing I can say is that the extra flicker in those blank 16 rows is gone. Is this where I need to experiment with led-multiplexing? I'm using 1. 

Should I try soldering E to 16? If I have solder on 8 and 16 from E will that cause issues?

Upgrade the board to 4x resolution they said... it'll be an easy swap out they said...

**Reply by BillLehecka** [reply]
OK... more info. The led board is HUB75E. So would it be fair to say I need to do something special for this since I believe that means it's ABCDE? Or did I already do that with the E and 8 solder job?

EDIT: OK, the more and more I'm seeing here is that my issue is that I'm using the Adafruit Bonnet that works with HUB75, but my panel is HUB75E. Can I retrofit the Adafruit Bonnet to accept HUB75E? Or should I just go with a different bonnet? It's a shame Adafruit doesn't have a HUB75E bonnet available. 

I guess my question is now that I soldered E to 8 on the bottom of the bonnet; can I do more so we can actually read the E line?

**Reply by ty-porter** [reply]
Let's see a picture of the solder job.

[This picture](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/discussions/495#discussioncomment-6631807), for instance, shows what looks like a cold solder joint on pin 18 and no solder at all on pin 4?

https://learn.adafruit.com/adafruit-guide-excellent-soldering/common-problems

**Reply by BillLehecka** [reply]
Oh OK. Just as an aside, this was my first solder job in 2020 where I wanted to learn something during pandemic times, so be gentile. I fully understand if I FUBARed the bonnet, I'll need a new one. 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/8aebf319-6a2f-4eba-a767-9c288a10687c)

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/c303e91e-150b-41fc-8c55-07b353085fcf)

Pin 18, for some reason, got more solder on the front, whereas pin 4 got more on the back which is why 4 looks bare. Not sure what 3 years past me was thinking... it worked with the 64x32 board so I thought nothing of it. It had worked beautifully for 3 years, but Present Bill wanted to upgrade the board after he made some new 3D printed flair to it as a salve for how horribly the Mets performed this season... but I digress. 

The E->8 solder job was the one I did yesterday and got overzealous which looks like it covers both. I assume this is no bueno, but the screen didn't malfunction worse, so I think we're Ok? 

I'm wondering if I need to run something from the HUB75 port on the bonnet to a pin on the breakout just so it can access the E line, in a way hacking the HUB75 connector on the bonnet and making it a HUB75E connector. Or is that way above my paygrade?

If anything, I hope I work through this enough so that when people Google this issue, like I have tried, they'll find this and we can hopefully find either a workable solution or a bonnet that would be better suited for this HUB75E board.

**Reply by ty-porter** [reply]
This is certainly your issue -- this is not an acceptable solder job. Shorting random pads on your bonnet is very unlikely to work.

You can fix this, but I advise you look up a soldering tutorial to proceed. You are not getting your iron hot enough and it is not wicking solder where it needs to go. Solder joints should cover the entire pad with no bridges and be shiny and uniform.

To fix this you might need some solder wick to clean up excess solder, but I think if you get the iron hot you can probably wick what's already there to ONLY bridge E -> 8.

FYI, the back of Pin 4 also looks like a short (same problems, too much solder, soldered cold, didn't wick). Pin 18 has the same problem on the front of the board.

**Comment by c2mfj** [comment]
Also look up how to 'tin' your solder tip. It makes a huge difference in
the heat transfer and if done correctly should melt the solder near
instantly.

**Comment by BillLehecka** [comment]
I appreciate the feedback, people. It's literally my first rodeo with this stuff. And I probably only got as far as I did by dumb luck. 

I watched tutorials on how to do it back in 2020, but apparently I needed more time to do it properly. I can say my soldering hasn't really improved all that much. But I definitely agree I should invest in a wick and probably should get a replacement bonnet in case I royally screw this up. I have some flux to tin somewhere in my work bench. 

I can work on fixing it up next week and definitely appreciate the notes. I'll report back then.

**Comment by BillLehecka** [comment]
Hey kids, sorry for the delay. Had some personal issues crop up that needed to be addressed. 

OK, decided to grab another Adafruit Bonnet and start from square one. 

First, I soldered 4 to 18. My soldering skill hasn't improved, but I verified success using my multimeter, so I don't think this is the issue. 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/033bdd78-1357-4e4e-8541-4a15273cf1c1)

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/513f9590-0aeb-4e65-9056-1149fc8caa80)

I then soldered E to 8. This was a much better job. 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/cd566f81-544a-49c3-8101-cbc026ea85e7)

I'm still getting 16 lines on, 16 lines off. 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/ea3a33a6-ba96-406b-be32-3b87646c54ee)

I'm at the point where I either need to do more soldering, or my startup arguments need to change. 

My current startup arguments are:

SCOREBOARD_ARGS=--led-gpio-mapping=adafruit-hat-pwm --led-brightness=40 --led-slowdown-gpio=4 --led-scan-mode=1 --led-cols=128 --led-rows=64 --led-multiplexing=1 --led-row-addr-type=0

Aside from using a HUB75 instead of this new HUB75E board, I'm at a loss.

**Reply by ty-porter** [reply]
Frankly, I don't know how else to help you -- we do not make or sell the hardware nor do we maintain the [driver library for it](https://github.com/hzeller/rpi-rgb-led-matrix) (we just build MLB-specific code on top of all of it).

My advice would be to play with `led-multiplexing` arguments (they go from 0 to 18) and `led-row-addr-type` (from 0 to 4). Also try `led-gpio-mapping=adafruit-hat`.

I am not convinced your soldering is not an issue. Your iron is still too cold and solder is not wicking to the pads.

**Comment by BillLehecka** [comment]
I honestly think it's a HUB75 vs HUB75E issue.  Once I get the new board, I'll report back.

**Reply by pjockey** [reply]
![20230817_192441](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/1387701/4091fb3f-08e2-444f-8bd9-9f18511da29e)
![20230817_192451](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/1387701/86ed8e31-9fb7-454a-92a3-22ad9fb1beab)
![20230817_192451](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/1387701/05756965-124b-4da1-943c-fd3333bf95ea)
Below is my start up line for the board in the above images:
python3 main.py --led-gpio-mapping=adafruit-hat-pwm --led-brightness=60 --led-slowdown-gpio=2 --led-cols=128 --led-rows=64 --led-rgb-sequence=BGR --led-row-addr-type=3

**Comment by bierbarrel** [comment]
I never could get the adafruit card to support more than 64 in either direction. No matter what I did. I went a different driver board. But, with that said, Im using 4 boards in a 128 by 64 layout with them daisy chained together.

**Reply by BillLehecka** [reply]
May I ask what driver board you went with?

**Reply by bierbarrel** [reply]
Electrodragon....I think they are out of Hong Kong but they came quick and were cheap....work way better than adafruit card. I ordered 3 because they were so cheap.

https://www.electrodragon.com/product/rgb-matrix-panel-drive-board-for-raspberry-pi-v2/

**Reply by BillLehecka** [reply]
Hi @bierbarrel, so I finally got those Electrodragon matrix panel boards delivered today!

[image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/0c501675-03a6-4004-8bc5-71ebc3721310)

I also ordered 3. 

How do you have it set up? It's a little less plug and play than I'm used to, so running power to it as of this moment is foreign to me. Any feedback would be appreciated. 

I assume the connector for the GND/VIN needs to be soldered in. I am also choosing to solder in pins at E and 8 which is probably overkill but will allow me to experiment with those jumpers without committing to a direct wire. Do I need to get a barrel jack and add it to the board so I can run the whole setup at 5V?

**Reply by bierbarrel** [reply]
Hmmm...let me go take a pic of mine. It looks a bit different from yours. Gimme a couple minutes.

**Reply by bierbarrel** [reply]
![PXL_20230912_201050123.jpg](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/75229947/a8e5deb4-c949-478f-9615-e36a02530f6b)

**Reply by bierbarrel** [reply]
Mine plugs directly on the pi...It has no other connections. They make a couple different versions of the card. I think the one you have will work.

**Reply by BillLehecka** [reply]
Got it. So if I power the panel separate, instead of worrying having the current switching power supply powering the Pi and the panel, we should be good. 

Will report back tomorrow! Very exciting to be near the end of the road.

**Comment by bierbarrel** [comment]
I did use that card in the past. I dont think it needs external power but I could be wrong. I gave it to my son. With that said, I might have soldered the screw terminals on the board and applied power to the card to take the load off of the pi. Did you see the extra info for the card? I did not do anything with the e line.

https://w2.electrodragon.com/Board/MPC/MPC1119-DAT/MPC1119-DAT.md

**Comment by c2mfj** [comment]
I just set up a 64x32 with the v2 electrodragon and an rpi 2b. I soldered
on the power terminals to power the matrix and used a 4A micro usb power
adapter to the pi only and it runs well.

**Reply by bierbarrel** [reply]
Best I remember that is exactly what I did.

**Comment by BillLehecka** [comment]
OK! I finally got it working! 

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/330e0b3a-9fbe-41a3-a49a-d15b34e85e13)

That was more of a pain in the butt than I thought it would be. 

So, for SEO purposes and in case anyone else has my issue when they want to upgrade to a HUB75E 128x64 board and can't find a simple answer, I'll try my best to help:

The Adafruit Bonnet is not your friend here and will not really work with a HUB75E board unless you want to get deep into the weeds with electrical stuff. I am not an engineer nor am I an electrician, so do what I did: get an Electrodragon RGB Matrix Board. 

https://www.electrodragon.com/product/rgb-matrix-panel-drive-board-for-raspberry-pi-v2/

Yeah, it's from China. Yeah, it took about 3 weeks to get here. Yeah, they don't really have any in the states... but it is worth it. 

Since this was a HUB75E board I had, I needed a jumper. For me, putting a jumper on P8 and E_BUF did the trick. 

Before:

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/050de57d-2d8c-4a20-b693-61f3561285e2)

After:

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/5ecfa0e0-ed9d-4861-b81a-0dd369122398)

One of the cool things about the Adafruit bonnet was how you could just plug in the switching power supply and it would power the board and the Pi. Well, the Electrodragon hat doesn't really have that... so if you just get a 5.5mm/2.1mm female connector, you can still use the power supply you might have had with the Adafruit Hat/Bonnet, but you'll also want to independently power your Pi. Extra plug? Yeah. Could that be fixed? I'm sure it could, but like I said, I'm not an engineer and I'm not an electrician. So extra plug it is!

I then updated my startup arguments to the following: 

--led-brightness=50 --led-slowdown-gpio=2 --led-scan-mode=1 --led-cols=128 --led-rows=64 --led-multiplexing=0 --led-row-addr-type=0

I rebooted and voila, it worked!

![image](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/33526260/f6c7642a-226f-4425-a192-30e3855d2515)

Adding the Mets Scoreboard header and lights to it was the right amount of extra for this to stand out from other implementations. 

Hopefully this helps anyone else that may encounter this issue. 

Thanks again to the regulars for bearing with my newbie incompetence. I can write code. I can configure databases. I just don't have the talent to be an electrical engineer. 

EDIT: One other note: I got a flexible display. This doesn't play nice if you decide to 3D print the enclosure for this. However, this flex display has magnets, so I grabbed some magnets off of Amazon and attached them to the 3D printed enclosure. This now gives me easy access inside if I need to fix something. I just pop it off and have access to the Pi.

**Reply by MrDennisM** [reply]
Hello, how did you make the Mets Scoreboard header?  Is it 3D printed or something else?  Huge Cubs fan here in suburban Chicago.  Would love to have something like and light it up too!  Thanks for sharing your recipe!  Thanks

**Reply by BillLehecka** [reply]
I had a guy on Fiverr design a 3D model so I could print it at home with my 3D Printer. I then painted it. I wanted to make it more interactive, but maybe some other day.

**Reply by MrDennisM** [reply]
Hi Bill, Do you recall the Fiverr designer?  If so, could you forward to me?  I'd gladly pay you to 3D print it for me.  I have two RGB panels together.  I apologize to all in advance if I should not be asking this here in a post.

**Reply by BillLehecka** [reply]
I don't recall the guy. I can PM you. Or take it offline.

**Reply by MrDennisM** [reply]
Thanks Bill!

**Comment by c2mfj** [comment]
Nice work! are those solderless jumpers?

**Reply by BillLehecka** [reply]
I soldered pins to those spots so I could just plug in jumpers because I wasn't sure if 4 or 8 would do the trick when connecting to E. It gave me flexibility. If I didn't have the space, I would've just soldered a wire between 8 and E.

**Comment by bierbarrel** [comment]
That looks awesome! Except for one small little last place error.On Sep 14, 2023 12:36 PM, Bill Lehecka ***@***.***> wrote:
I soldered pins to those spots so I could just plug in jumpers because I wasn't sure if 4 or 8 would do the trick when connecting to E. It gave me flexibility. If I didn't have the space, I would've just soldered a wire between 8 and E.

**Reply by BillLehecka** [reply]
Hey now, the Mets are clearly in 4th place!!! LOL

I laugh because I'm all out of tears.

**Comment by bierbarrel** [comment]
Lol...too funny On Sep 14, 2023 2:11 PM, Bill Lehecka ***@***.***> wrote:
Hey now, the Mets are clearly in 4th place!!! LOL
I laugh because I'm all out of tears.

**Comment by pjockey** [comment]
> Adding the Mets Scoreboard header and lights to it was the right amount of extra for this to stand out from other implementations.

Do you have any instructions on this lighting?

**Reply by BillLehecka** [reply]
Sure. Get a light strip that is powered by USB. Plug it into your Pi. Voila. It'll turn on with your Pi.
